src/html/html-parser.ts

```diff
@@ -0,0 +1,80 @@
+/**
+ * @license
+ * Copyright (c) 2015 The Polymer Project Authors. All rights reserved.
+ * This code may only be used under the BSD style license found at http://polymer.github.io/LICENSE.txt
+ * The complete set of authors may be found at http://polymer.github.io/AUTHORS.txt
+ * The complete set of contributors may be found at http://polymer.github.io/CONTRIBUTORS.txt
+ * Code distributed by Google as part of the polymer project is also
+ * subject to an additional IP rights grant found at http://polymer.github.io/PATENTS.txt
+ */
+
+import * as dom5 from 'dom5';
+import * as parse5 from 'parse5';
+import {ASTNode} from 'parse5';
+
+import {Analyzer} from '../analyzer';
+import {Parser} from '../parser/parser';
+import {ImportDescriptor} from '../ast/ast';
+
+import {HtmlDocument} from './html-document';
+import {Document} from '../parser/document';
+
+export function getOwnerDocument(node: parse5.ASTNode): parse5.ASTNode {
+  while (node && !dom5.isDocument(node)) {
+    node = node.parentNode;
+  }
+  return node;
+}
+
+const p = dom5.predicates;
+
+const isInlineJSScript = p.AND(
+  p.hasTagName('script'),
+  p.NOT(p.hasAttr('src')),
+  p.OR(
+    p.NOT(p.hasAttr('type')),
+    p.hasAttrValue('type', 'text/javascript'),
+    p.hasAttrValue('type', 'application/javascript')
+  )
+);
```

Comment:
Oh wait, nm, we don't need to care about whether a given script is in module mode or not at parse time. Only time it would show up might be in inferring whether a toplevel identifier is global.